### DISCUSSION TITLE
How to send note/freq information to Faust

### DISCUSSION THREAD

**Discussion by jonrichings** [pschatzmann/arduino-audio-tools#219] [Q&A] [answered]
I understand sending faust.setLabelValue(); to change a control value.  I'm unclear about how to send note/freq.  I imagine that it is often just a case of finding the .dsp definition for the slider that controls the instruments frequency and setLabelValue that control.  But that isn't always clear.  For instance the Faust flute example does not have a main frequency control (as far as I can see).  

Many of the Faust examples need a gate input to trigger even when there is no frequency as such.  How do you trigger a gate?

And does AudioFaust support .dsp with MIDI?  Would it be possible (easier) to just send MIDI data to the .dsp?

**Comment by pschatzmann** [comment, marked as answer]
Not all Faust examples support midi - the Faust flute is an example and you control the frequency with the pipe length.
Supporting Faust Midi is currently out of scope and there are enough midi libraries for Arduino to choose from. 
You would control the freqeuncy with setValue("freq", freqValue);

For convenience I have added some additional methods to make it less tedious and I let you test them:

- setFrequency
- setMidiNote
- setBend
- setGain
- midiOn(int note, FAUSTFLOAT gain)
- midiOff(int note)

**Reply by jonrichings** [reply]
Nice.  I am not quite there yet but I can see how those will help a lot.

**Comment by sletz** [comment]
The Faust MIDI/Polyphonic model is explained https://faustdoc.grame.fr/manual/midi/#midi-polyphony-support and https://faustdoc.grame.fr/manual/architectures/#polyphonic-instruments

**Reply by jonrichings** [reply]
Hopefully that and Phil's additions will make poly possible.  I'm getting there...
